conformance/lib/src/jsMain/kotlin/pbandk/conformance/Platform.kt

```diff
@@ -1,50 +1,107 @@
 package pbandk.conformance
 
+import org.khronos.webgl.ArrayBufferView
+import org.khronos.webgl.Uint8Array
+import pbandk.asByteArray
 import pbandk.asUint8Array
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
 
-external fun require(module: String): dynamic
-external class Buffer {
+internal external interface StdStream {
+    val fd: Int
+
+    fun once(event: String, callback: () -> Unit)
+
+    // Readable streams only
+    fun read(size: Int = definedExternally): Buffer?
+
+    // Writeable streams only
+    fun write(chunk: String, encoding: String = definedExternally): Boolean
+    fun write(chunk: Uint8Array): Boolean
+}
+
+@JsModule("process")
+internal external class Process {
+    companion object {
+        val stdin: StdStream
+        val stdout: StdStream
+        val stderr: StdStream
+    }
+}
+
+@JsModule("fs")
+external class Fs {
+    companion object {
+        fun writeSync(
+            fd: Int,
+            buffer: ArrayBufferView,
+            offset: Int,
+            length: Int,
+            position: Int? = definedExternally
+        ): Int
+    }
+}
+
+external class Buffer : Uint8Array {
     fun readInt32LE(offset: Int): Int
     fun writeInt32LE(value: Int, offset: Int): Int
+
     companion object {
         fun alloc(size: Int): Buffer
     }
 }
 
 actual object Platform {
-    val process = require("process")
-    val fs = require("fs")
-
-    actual fun stderrPrintln(str: String) { process.stderr.write("$str\n") }
+    actual fun stderrPrintln(str: String) {
+        Process.stderr.write("$str\n")
+    }
 
-    fun stdinReadBuffer(buf: dynamic): Boolean {
+    private fun stdinReadNow(size: Int): Buffer? {
```

Comment:
How would you know the size to read from stdin before calling this function?

Reply:
It's a few steps removed from here:
1. This function is just a helper for `stdinReadBuffer(size)`
2. `stdinReadBuffer(size)` is called from `stdinReadIntLE()` and `stdinReadFull(size)`, both of which know the number of bytes they need to read
3. `stdinReadFull(size)` is used to read conformance request objects on stdin from the `conformance-test-runner`. The `conformance-test-runner` prefixes each object with its length, so we always know how many bytes we'll need to read from stdin when we get a new conformance request.